test/behaviour/typeql/TypeQLSteps.java

```diff
@@ -693,35 +695,33 @@ public boolean check(Concept concept) {
                 return false;
             }
 
-            Set<Attribute<?>> keys = concept.asThing().asRemote(tx()).getHas(true).collect(Collectors.toSet());
-            HashMap<Label, String> keyMap = new HashMap<>();
-
-            for (Attribute<?> key : keys) {
-                String keyValue;
-                switch (key.getType().getValueType()) {
-                    case BOOLEAN:
-                        keyValue = key.asBoolean().getValue().toString();
-                        break;
-                    case LONG:
-                        keyValue = key.asLong().getValue().toString();
-                        break;
-                    case DOUBLE:
-                        keyValue = key.asDouble().getValue().toString();
-                        break;
-                    case STRING:
-                        keyValue = key.asString().getValue();
-                        break;
-                    case DATETIME:
-                        keyValue = key.asDateTime().getValue().toString();
-                        break;
-                    case OBJECT:
-                    default:
-                        throw new ScenarioDefinitionException("Unrecognised value type " + key.getType().getValueType());
-                }
-
-                keyMap.put(key.getType().getLabel(), keyValue);
+            Optional<? extends Attribute<?>> opt_key = concept.asThing().asRemote(tx()).getHas(true)
```

Comment:
In Java we don't use `_` for variable names, it's camelCase. Also I would call this `keyOpt` instead of `optKey` I think